hilti/toolchain/src/ast/scope-lookup.cc

```diff
@@ -11,6 +11,9 @@
 using namespace hilti;
 
 std::pair<bool, Result<std::pair<Declaration*, ID>>> hilti::scope::detail::lookupID(const ID& id, const Node* n) {
+    if ( ! n || ! n->scope() )
+        return {false, result::Error("cannot perform lookup if node or scope is missing")};
+
```

Comment:
Couple things:

- `scope == nullptr` is generally expected: it's the same as an empty scope. Scope objects won't be allocated unless they are needed to actually store something. With that, I'd change the error message to the same "unknown ID" that we use further below if we don't find the ID. (I'm not quite sure why we haven't seen this before given that it's not unusual; I guess we have null checks elsewhere).
- `n == nullptr` shouldn't really happen, I'd do an `assert` or `internalError` there.